### DISCUSSION TITLE
getting lwipopts.h not found while compiling on raspberry pi for teensy 4.1

### DISCUSSION THREAD

**Discussion by jrdarrah** [terjeio/grblHAL#193] [Q&A]
I'm trying to compile grblHAL tor the Teensy 4 on the grblHAL breakout board to include ethernet support usinng the Arduino IDE on a Raspberry PI 4. While I've made  some progress I'm now getting an error that lwipopts.h is not found. the offending statement is in enet.c. Do I need to include a specific library.  In doing a find, aside from two grblHAL processor directoried I only see the file in the Arduino Wifi library. I told it to use the Wifi library and still get the error.  Any information on resolving the problemm will be greatly  appreciated. 

/home/pi/arduino-1.8.13/arduino-builder -dump-prefs -logger=machine -hardware /home/pi/arduino-1.8.13/hardware -tools /home/pi/arduino-1.8.13/tools-builder -tools /home/pi/arduino-1.8.13/hardware/tools/avr -built-in-libraries /home/pi/arduino-1.8.13/libraries -libraries /home/pi/Arduino/libraries -fqbn=teensy:avr:teensy41:usb=serial,speed=600,opt=o2std,keys=en-us -ide-version=10813 -build-path /tmp/arduino_build_50769 -warnings=none -build-cache /tmp/arduino_cache_877125 -verbose /home/pi/Arduino/libraries/grblHAL_Teensy4/examples/grblHAL_Teensy4_Upload/grblHAL_Teensy4_Upload.ino
/home/pi/arduino-1.8.13/arduino-builder -compile -logger=machine -hardware /home/pi/arduino-1.8.13/hardware -tools /home/pi/arduino-1.8.13/tools-builder -tools /home/pi/arduino-1.8.13/hardware/tools/avr -built-in-libraries /home/pi/arduino-1.8.13/libraries -libraries /home/pi/Arduino/libraries -fqbn=teensy:avr:teensy41:usb=serial,speed=600,opt=o2std,keys=en-us -ide-version=10813 -build-path /tmp/arduino_build_50769 -warnings=none -build-cache /tmp/arduino_cache_877125 -verbose /home/pi/Arduino/libraries/grblHAL_Teensy4/examples/grblHAL_Teensy4_Upload/grblHAL_Teensy4_Upload.ino
Using board 'teensy41' from platform in folder: /home/pi/arduino-1.8.13/hardware/teensy/avr
Using core 'teensy4' from platform in folder: /home/pi/arduino-1.8.13/hardware/teensy/avr
Detecting libraries used...
/home/pi/arduino-1.8.13/hardware/teensy/../tools/arm/bin/arm-none-eabi-g++ -E -CC -x c++ -w -g -Wall -ffunction-sections -fdata-sections -nostdlib -std=gnu++14 -fno-exceptions -fpermissive -fno-rtti -fno-threadsafe-statics -felide-constructors -Wno-error=narrowing -mthumb -mcpu=cortex-m7 -mfloat-abi=hard -mfpu=fpv5-d16 -D__IMXRT1062__ -DTEENSYDUINO=153 -DARDUINO=10813 -DARDUINO_TEENSY41 -DF_CPU=600000000 -DUSB_SERIAL -DLAYOUT_US_ENGLISH -I/home/pi/arduino-1.8.13/hardware/teensy/avr/cores/teensy4 /tmp/arduino_build_50769/sketch/grblHAL_Teensy4_Upload.ino.cpp -o /dev/null -DARDUINO_LIB_DISCOVERY_PHASE
Alternatives for grblHAL_Teensy4.h: [grblHAL_Teensy4@1.1.7]
ResolveLibrary(grblHAL_Teensy4.h)
  -> candidates: [grblHAL_Teensy4@1.1.7]
/home/pi/arduino-1.8.13/hardware/teensy/../tools/arm/bin/arm-none-eabi-g++ -E -CC -x c++ -w -g -Wall -ffunction-sections -fdata-sections -nostdlib -std=gnu++14 -fno-exceptions -fpermissive -fno-rtti -fno-threadsafe-statics -felide-constructors -Wno-error=narrowing -mthumb -mcpu=cortex-m7 -mfloat-abi=hard -mfpu=fpv5-d16 -D__IMXRT1062__ -DTEENSYDUINO=153 -DARDUINO=10813 -DARDUINO_TEENSY41 -DF_CPU=600000000 -DUSB_SERIAL -DLAYOUT_US_ENGLISH -I/home/pi/arduino-1.8.13/hardware/teensy/avr/cores/teensy4 -I/home/pi/Arduino/libraries/grblHAL_Teensy4/src /tmp/arduino_build_50769/sketch/grblHAL_Teensy4_Upload.ino.cpp -o /dev/null -DARDUINO_LIB_DISCOVERY_PHASE
/home/pi/arduino-1.8.13/hardware/teensy/../tools/arm/bin/arm-none-eabi-g++ -E -CC -x c++ -w -g -Wall -ffunction-sections -fdata-sections -nostdlib -std=gnu++14 -fno-exceptions -fpermissive -fno-rtti -fno-threadsafe-statics -felide-constructors -Wno-error=narrowing -mthumb -mcpu=cortex-m7 -mfloat-abi=hard -mfpu=fpv5-d16 -D__IMXRT1062__ -DTEENSYDUINO=153 -DARDUINO=10813 -DARDUINO_TEENSY41 -DF_CPU=600000000 -DUSB_SERIAL -DLAYOUT_US_ENGLISH -I/home/pi/arduino-1.8.13/hardware/teensy/avr/cores/teensy4 -I/home/pi/Arduino/libraries/grblHAL_Teensy4/src /home/pi/Arduino/libraries/grblHAL_Teensy4/src/driver.c -o /dev/null -DARDUINO_LIB_DISCOVERY_PHASE
/home/pi/arduino-1.8.13/hardware/teensy/../tools/arm/bin/arm-none-eabi-g++ -E -CC -x c++ -w -g -Wall -ffunction-sections -fdata-sections -nostdlib -std=gnu++14 -fno-exceptions -fpermissive -fno-rtti -fno-threadsafe-statics -felide-constructors -Wno-error=narrowing -mthumb -mcpu=cortex-m7 -mfloat-abi=hard -mfpu=fpv5-d16 -D__IMXRT1062__ -DTEENSYDUINO=153 -DARDUINO=10813 -DARDUINO_TEENSY41 -DF_CPU=600000000 -DUSB_SERIAL -DLAYOUT_US_ENGLISH -I/home/pi/arduino-1.8.13/hardware/teensy/avr/cores/teensy4 -I/home/pi/Arduino/libraries/grblHAL_Teensy4/src /home/pi/Arduino/libraries/grblHAL_Teensy4/src/enet.c -o /dev/null -DARDUINO_LIB_DISCOVERY_PHASE
Alternatives for lwipopts.h: []
ResolveLibrary(lwipopts.h)
  -> candidates: []
/home/pi/Arduino/libraries/grblHAL_Teensy4/src/enet.c:30:22: fatal error: lwipopts.h: No such file or directory
compilation terminated.
Using library grblHAL_Teensy4 at version 1.1.7 in folder: /home/pi/Arduino/libraries/grblHAL_Teensy4 
Error compiling for board Teensy 4.1.

**Comment by terjeio** [comment]
You need a ethernet library, info/link on the [driver page](https://github.com/terjeio/grblHAL/tree/master/drivers/IMXRT1062).
